staking/integration-tests/tests/stability.rs

```diff
@@ -0,0 +1,833 @@
+        init_pool_data:  true,
+        init_publishers: false,
+    });
+
+    props.delegators.iter().for_each(|delegator| {
+        svm.airdrop(&delegator.0.pubkey(), 1_000_000_000).unwrap();
+    });
+
+    let stake_account_positions: Vec<Pubkey> = props
+        .delegators
+        .iter()
+        .enumerate()
+        .map(|(i, delegator)| {
+            if i % 100 == 0 {
+                println!(
+                    "creating stake account {} out of {}",
+                    i,
+                    props.delegators.len()
+                );
+            }
+            initialize_new_stake_account(&mut svm, &delegator.0, &pyth_token_mint, true, true)
+        })
+        .collect();
+
+    initialize_pool_reward_custody(&mut svm, &payer, &pyth_token_mint);
+
+    update_y(&mut svm, &payer, &reward_program_authority, Y).unwrap();
+
+    let publisher_caps = post_publisher_caps(
+        &mut svm,
+        &payer,
+        props.publishers.iter().map(|p| p.0.pubkey()).collect(),
+        vec![0; props.publishers.len()],
+    );
+    advance(&mut svm, &payer, publisher_caps).unwrap();
+
+    let mut operation_counts: HashMap<String, (u64, u64)> = HashMap::new();
+
+    operation_counts.insert(
+        "Delegate".to_string(),
+        (
+            0,
+            props
+                .operations
+                .iter()
+                .filter(|op| matches!(op, Operation::Delegate { .. }))
+                .count() as u64,
+        ),
+    );
+    operation_counts.insert(
+        "Undelegate".to_string(),
+        (
+            0,
+            props
+                .operations
+                .iter()
+                .filter(|op| matches!(op, Operation::Undelegate { .. }))
+                .count() as u64,
+        ),
+    );
+    operation_counts.insert(
+        "AdvanceDelegationRecord".to_string(),
+        (
+            0,
+            props
+                .operations
+                .iter()
+                .filter(|op| matches!(op, Operation::AdvanceDelegationRecord { .. }))
+                .count() as u64,
+        ),
+    );
+    operation_counts.insert(
+        "Advance".to_string(),
+        (
+            0,
+            props
+                .operations
+                .iter()
+                .filter(|op| matches!(op, Operation::Advance { .. }))
+                .count() as u64,
+        ),
+    );
+    operation_counts.insert(
+        "Slash".to_string(),
+        (
+            0,
+            props
+                .operations
+                .iter()
+                .filter(|op| matches!(op, Operation::Slash { .. }))
+                .count() as u64,
+        ),
+    );
+    operation_counts.insert(
+        "MergePositions".to_string(),
+        (
+            0,
+            props
+                .operations
+                .iter()
+                .filter(|op| matches!(op, Operation::MergePositions { .. }))
+                .count() as u64,
+        ),
+    );
+    operation_counts.insert(
```

Comment:
I don't understand why you insert here and then later in the for loop too

Reply:
also you could use get_name here too

Reply:
improved it.